From da5d48accb9bce65441827645ee0b06afccb5732 Mon Sep 17 00:00:00 2001
From: Javier Pena <jpena@redhat.com>
Date: Wed, 20 May 2015 11:32:18 +0200
Subject: [PATCH] Fix xmldsig import

After commit https://goo.gl/TjLQFz to pysaml2, xmldsig was moved
under saml2, no longer being at the top level. Craeting a
conditional import, so it can work now and when the pysaml2
commit is included in a new release.

Conflicts:
	keystone/tests/unit/test_v3_federation.py

Change-Id: If01a93a18cef6159adad50496e6335e88f86024d
(cherry picked from commit 71903b78d6b57877c32319335f7aef87332585e9)
(cherry picked from commit d4dcef5098b9f10a239449e05301a7db8410cdb5)
---
 keystone/contrib/federation/idp.py        | 5 ++++-
 keystone/tests/unit/test_v3_federation.py | 5 ++++-
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/keystone/contrib/federation/idp.py b/keystone/contrib/federation/idp.py
index 4bca858..84b9332 100644
--- a/keystone/contrib/federation/idp.py
+++ b/keystone/contrib/federation/idp.py
@@ -17,6 +17,7 @@ import uuid
 
 from oslo_config import cfg
 from oslo_log import log
+from oslo_utils import importutils
 from oslo_utils import timeutils
 import saml2
 from saml2 import client_base
@@ -26,7 +27,9 @@ from saml2 import saml
 from saml2 import samlp
 from saml2.schema import soapenv
 from saml2 import sigver
-import xmldsig
+xmldsig = importutils.try_import("xmldsig")
+if not xmldsig:
+    xmldsig = importutils.try_import("saml2.xmldsig")
 
 from keystone import exception
 from keystone.i18n import _, _LE
diff --git a/keystone/tests/unit/test_v3_federation.py b/keystone/tests/unit/test_v3_federation.py
index c1a4a67..efc7649 100644
--- a/keystone/tests/unit/test_v3_federation.py
+++ b/keystone/tests/unit/test_v3_federation.py
@@ -21,12 +21,15 @@ import mock
 from oslo_config import cfg
 from oslo_log import log
 from oslo_serialization import jsonutils
+from oslo_utils import importutils
 from oslotest import mockpatch
 import saml2
 from saml2 import saml
 from saml2 import sigver
 from six.moves import urllib
-import xmldsig
+xmldsig = importutils.try_import("xmldsig")
+if not xmldsig:
+    xmldsig = importutils.try_import("saml2.xmldsig")
 
 from keystone.auth import controllers as auth_controllers
 from keystone.auth.plugins import mapped
